### PR TITLE
Add section about Short Circuit Evaluation

### PR DESCRIPTION
**Pull request by vitorlopez**
Add section about Short Circuit Evaluation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -593,7 +593,38 @@
...
+
+   // bad
+   return someBooleanVar && someFunction() || 'defaultValue';
+
+    // good
+    if (someBooleanVar) {
+       return someFunction();
+    }
+    return 'defaultValue';
+
```
Comment (goatslacker) [inline README.md]: forgot a closing triple backtick.

File: README.md [hunk 2]
```diff
@@ -593,7 +593,38 @@
     }
     ```
 
-  - For more information see [Truth Equality and JavaScript](http://javascriptweblog.wordpress.com/2011/02/07/truth-equality-and-javascript/#more-2108) by Angus Croll
+- Avoid Short Circuit Evaluation
+
+    ```javascript
```
Comment (goatslacker) [inline README.md]: you can shorten this to just js

**Comment by hshoff** [comment]
Once @goatslacker's comments are addressed, this LGTM.

**Comment by ljharb** [comment]
@vlopez5 this is great - abusing value selection operators for control flow is bad news bears indeed. Are you still interested in landing this? It'd be great if you could rebase on latest master and address the comments in the PR :-)

**Comment by schinsue** [comment]
Please merge this in! :(

**Comment by ljharb** [comment]
@schinsue it's waiting on https://github.com/airbnb/javascript/pull/188#issuecomment-166979277 and/or the original PR author checking the "allow edits" box on the right side of the PR.

**Comment by schinsue** [comment]
@ljharb Am I allowed to create a fork and follow up on the comments made in the original PR?

**Comment by vlopez5** [comment]
I'm sorry for the delay guys... I already checked the box to allow edits! Thanks guys!

**Comment by ljharb** [comment]
I've rebased this; but indeed, https://github.com/airbnb/javascript#control-statements--value-selection seems to already cover it (#1729)
